src/vivarium_public_health/risks/effect.py

```diff
@@ -316,3 +320,164 @@ def _get_risk_exposure_class(self, builder: Builder) -> Risk:
+            rr_data = pd.concat([df, max_exposure_row]).reset_index()
+            rr_data["left_exposure"] = [0] + rr_data["parameter"][:-1].tolist()
+            rr_data["left_rr"] = [rr_data["value"].min()] + rr_data["value"][:-1].tolist()
+            rr_data["right_exposure"] = rr_data["parameter"]
+            rr_data["right_rr"] = rr_data["value"]
+            return rr_data[
+                ["parameter", "left_exposure", "left_rr", "right_exposure", "right_rr"]
+            ]
+
+        # define exposure and rr interval columns
+        demographic_cols = [
+            col for col in rr_data.columns if col != "parameter" and col != "value"
+        ]
+        rr_data = (
+            rr_data.groupby(demographic_cols)
+            .apply(define_rr_intervals)
+            .reset_index(level=-1, drop=True)
+            .reset_index()
+        )
+        rr_data = rr_data.drop("parameter", axis=1)
+        rr_data[f"{self.risk.name}_exposure_start"] = rr_data["left_exposure"]
+        rr_data[f"{self.risk.name}_exposure_end"] = rr_data["right_exposure"]
+        # build lookup table
+        rr_value_cols = ["left_exposure", "left_rr", "right_exposure", "right_rr"]
+        self.lookup_tables["relative_risk"] = self.build_lookup_table(
+            builder, rr_data, rr_value_cols
+        )
+
+        # paf_data = self.get_filtered_data(
+        #    builder, self.configuration.data_sources.population_attributable_fraction
+        # )
+        # self.lookup_tables["population_attributable_fraction"] = self.build_lookup_table(
+        #    builder, paf_data
+        # )
+        self.lookup_tables["population_attributable_fraction"] = builder.lookup.build_table(1)
+
+    def get_relative_risk_data(
+        self,
+        builder: Builder,
+        configuration=None,
+    ) -> Union[str, float, pd.DataFrame]:
+        if configuration is None:
+            configuration = self.configuration
+
+        # get TMREL
+        tmred = builder.data.load(f"{self.risk}.tmred")
+        if tmred["distribution"] == "uniform":
+            self.tmrel = np.random.uniform(tmred["min"], tmred["max"])
+        elif tmred["distribution"] == "draws":  # currently only for iron deficiency
+            raise MissingDataError(
+                f"TMRED has a non-uniform distribution. You will need to contact the research team that models {self.risk.name} to get this data."
+            )
+        else:
+            raise MissingDataError(f"No TMRED found in gbd_mapping for risk {self.risk.name}")
+
+        # calculate RR at TMREL
+        rr_source = configuration.data_sources.relative_risk
+        original_rrs = self.get_filtered_data(builder, rr_source)
+        demographic_cols = [
+            col for col in original_rrs.columns if col != "parameter" and col != "value"
+        ]
+
+        def get_rr_at_tmrel(rr_data: pd.DataFrame) -> float:
+            interpolated_rr_function = scipy.interpolate.interp1d(
```

Comment:
It works fine in this case because we only need to return a single number, but down below we would need to return the actual interpolated function. I could have created a lookup table that returns a hexadecimal representation of the interpolated function for each demographic group and use that to evaluate the exposures of each sub-group but this way seems simpler.